kimchi/src/folding/error.rs

```diff
@@ -0,0 +1,331 @@
+use crate::folding::{
+    expressions::{Degree, ExtendedFoldingColumn, FoldingExp, IntegratedFoldingExpr},
+    quadricization::ExtendedWitnessGenerator,
+    EvalLeaf, FoldingConfig, FoldingEnv, RelaxedInstance, RelaxedWitness,
+};
+use ark_ec::AffineCurve;
+use ark_ff::Field;
+use ark_poly::{Evaluations, Radix2EvaluationDomain};
+use poly_commitment::SRS;
+
+#[derive(Clone, Copy)]
+pub enum Side {
+    Left,
+    Right,
+}
+impl std::ops::Not for Side {
```

Comment:
This feels somewhat unexpected, it might be clearer to have an associated function `Side::other`